microsoft/testsuites/mshv/mshv_root_tests.py

```diff
@@ -252,6 +260,62 @@ def verify_mshv_crash(
         finally:
             node.mark_dirty()
 
+    @TestCaseMetadata(
+        description="""
+        Ensure mshvtrace tool is present, can be executed, and produces
+        output on the MSHV root partition.
+        The test checks:
+        1. mshvtrace binary exists and is executable
+        2. Running mshvtrace with --help returns expected output
+        3. Running mshvtrace for a short duration produces a non-empty trace file
+        """,
+    )
+    def verify_mshvtrace_tool(
+        self,
+        log: Logger,
+        node: Node,
+    ) -> None:
+        mshvtrace_path = "/usr/sbin/mshvtrace"
+        temp_dir = node.execute(
+            "mktemp -d /tmp/mshvtrace_test_XXXXXX", sudo=True
+        ).stdout.strip()
+        trace_output = f"{temp_dir}/mshvtrace_output.ETL"
+
+        try:
+            # 1. Check if mshvtrace exists and is executable
+            exists = node.tools[Ls].path_exists(mshvtrace_path, sudo=True)
+            assert_that(exists).described_as("mshvtrace binary should exist").is_true()
+            is_executable = (
+                node.execute(f"test -x {mshvtrace_path}", sudo=True).exit_code == 0
+            )
+            assert_that(is_executable).described_as(
+                "mshvtrace should be executable"
+            ).is_true()
+
+            # 2. Create tracing context
+            create_result = node.execute(f"{mshvtrace_path} create", sudo=True)
+            assert_that(create_result.exit_code).is_equal_to(0)
+
+            # 3. Collect trace for 30 seconds using node.execute with timeout argument
+            _ = node.execute(
+                f"{mshvtrace_path} collect -o {trace_output}", sudo=True, timeout=35
+            )
+            trace_size = node.execute(
+                f"stat -c %s {trace_output}", sudo=True
```

Comment:
There is already a `Stat` tool in lisa. Could you check if this can be done from the tool itself? If not, its good to add this new functionality in tool itself.

Reply:
make sense to use the Stat tool in lisa. I will refactor this

Reply:
updated.